pytplot/QtPlotter/TVarFigure1D.py

```diff
@@ -105,11 +106,95 @@ def _visdata(self):
         else:
             datasets.append(pytplot.data_quants[self.tvar_name])
         line_num = 0
+
         for dataset in datasets:
-            for i in range(0, len(dataset.data.columns)):
-                self.curves.append(self.plotwindow.plot(dataset.data.index.tolist(),
-                                                        dataset.data[i].tolist(),
-                                                        pen=self.colors[line_num % len(self.colors)]))
+            for i in range(len(dataset.data.columns)):
+                limit = pytplot.tplot_opt_glob['data_gap']  # How big a data gap is allowed before those nans (default
+                # is to plot as pyqtgraph would normally plot w / o worrying about data gap handling).
+                if limit != 0:
+                    index = 0  # What index in the below for loop we're on
+                    nans = dict()  # Where nan value indices + times associated with those indices are housed
+                    for j in dataset.data[i].keys():
+                        point = dataset.data[i][j]  # Just grab the point in that dataset, at a specific time
+                        if np.isnan(point):
+                            nans[index] = j  # If we have a nan value, put the index/time in the nan dictionary
+                        index += 1
+                    # Putting the keys & values from the nans dictionary to lists to make them more easily accesible
+                    nan_keys = list(nans.keys())
+                    nan_values = list(nans.values())
```

Comment:
You could replace the previous ~10 lines with:

dataset.data[i][dataset.data[i].isnull().values].index

Not sure if it is faster or not, any I haven't looked into how you would get the keys yet

Reply:
Not opposed to getting rid of this loop at all. haha I'll think about how to combine that idea and the keys/values to make it happen....

Reply:
Turns out another list comprehension is the answer for keys.